accounts/queries.py

```diff
@@ -0,0 +1,277 @@
+              | relation_expr
+?relation_expr: id comparator (NUMBER | TRUE | FALSE | NULL | id)
+
+?comparator: EQ | NE | LT | LTE | GT | GTE
+
+not_bool_factor: "NOT" bool_factor
+
+id: NAME
+
+// TERMINALS
+
+EQ: "="
+NE: "!="
+LT: "<"
+LTE: "<="
+GT: ">"
+GTE: ">="
+
+TRUE: "true"
+FALSE: "false"
+NULL: "null"
+
+%import common.NUMBER
+%import common.CNAME -> NAME
+%import common.WS
+%ignore WS
+"""
+
+QUERY_DSL_PARSER = Lark(QUERY_GRAMMAR, parser="earley")
+
+
+def get_child_eligibility_for_study(child_obj, study_obj):
+    return get_child_eligibility(child_obj, study_obj.criteria_expression)
+
+
+def get_child_eligibility(child_obj, criteria_expr):
+    if criteria_expr:
+        compiled_tester_func = compile_expression(criteria_expr)
+
+        expanded_child = _get_expanded_child(child_obj)
+
+        return bool(compiled_tester_func(expanded_child))
+    else:
+        return True
+
+
+def compile_expression(boolean_algebra_expression: str):
+    """Compiles a boolean algebra expression into a python function.
+
+    Args:
+        boolean_algebra_expression: a string boolean algebra expression.
+
+    Returns:
+        A function.
+
+    Raises:
+        lark.exceptions.ParseError: in case we cannot parse the boolean algebra.
+    """
+    parse_tree = QUERY_DSL_PARSER.parse(boolean_algebra_expression)
+
+    func_body = FunctionTransformer().transform(parse_tree)
+
+    func_text = " ".join(["def property_tester(child_obj):  return", func_body])
+
+    code_object = ast.parse(func_text, mode="exec")
+
+    new_func = compile(code_object, filename="temp.py", mode="exec")
+
+    temp_namespace = {}
+
+    exec(new_func, temp_namespace)
+
+    return temp_namespace["property_tester"]
+
+
+def _get_expanded_child(child_object):
+    """Expands a child object such that it can be evaluated easily.
+
+    The output of this method should be such that _compile_expression
+    can evaluate it; i.e. all keys are first-level.
+
+    Args:
+        child_object: a accounts.models.Child instance.
+
+    Returns:
+        A dict representing the child.
+    """
+    expanded_child = _to_dict(child_object)
+
+    # 1) Change birthday to age in days.
+    age_delta = date.today() - expanded_child.pop("birthday")
+    expanded_child["age_in_days"] = age_delta.days
+
+    # 2) Expand existing conditions in-place.
+    expanded_conditions = dict(expanded_child.pop("existing_conditions").items())
+    expanded_child.update(expanded_conditions)
+
+    # 3) Expand languages in place.
+    expanded_languages = {
+        f"speaks_{langcode}": boolean
+        for langcode, boolean in expanded_child.pop("languages_spoken").items()
+    }
+    expanded_child.update(expanded_languages)
+
+    ga_enum = expanded_child.pop("gestational_age_at_birth")
+
+    gestational_age_in_weeks = _gestational_age_enum_value_to_weeks(ga_enum)
+
+    expanded_child["gestational_age_in_weeks"] = gestational_age_in_weeks
+
+    return expanded_child
```

Comment:
Can we add n_languages here?

Reply:
That will be slightly more tricky but I think I can do it

Reply:
wouldn't it just be something like `expanded_child["n_languages"] = sum([bool for (lang, bool) in expanded_languages.items()])`?

Reply:
Got it done.

Reply:
did something similar

Reply:
Ohhh haha I see it was not as simple as adding another field at all. Thanks!

Reply:
Your way would have worked too - we'd just need to still add the grammar rule in the SDT